### PR TITLE
Derive supplementary output paths using the full basename of .swiftmodule

### PR DESCRIPTION
**Pull request by nkcsgexi**
rdar://70584224

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -2179,7 +2179,7 @@ extension Driver {
         parentPath = moduleOutputPath.parentDirectory
       }
 
-      return parentPath.appending(component: moduleName).replacingExtension(with: type)
+      return parentPath.appending(component: moduleOutputPath.basename).replacingExtension(with: type)
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Driver/Driver.swift:2182]: This function doesn't need to take `moduleName` as a parameter anymore.

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 2]
```diff
@@ -2529,6 +2529,21 @@ final class SwiftDriverTests: XCTestCase {
     }
   }
 
+  func testDeriveSwiftDocPath() throws {
+    var driver = try Driver(args: [
+      "swiftc", "-emit-module", "/tmp/main.swift", "-emit-module-path", "test-ios-macabi.swiftmodule"
```
Comment (cltnschlosser) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:2534]: What is module name in this case? main? Just wondering how off it was.

Comment (nkcsgexi) [reply Tests/SwiftDriverTests/SwiftDriverTests.swift:2534]: I think `main` will be the default name if not given at all.

File: Sources/SwiftDriver/Driver/Driver.swift (hunk 1 above)
Comment (nkcsgexi) [reply Sources/SwiftDriver/Driver/Driver.swift:2182]: It still needs `moduleName` in the last return statement

Comment (cltnschlosser) [reply Sources/SwiftDriver/Driver/Driver.swift:2182]: Oh oops, just looked up, thought this was the last one.

**Pull request merged**